[wjunjie01] iP

wjunjie01 opened the pull request.

wjunjie01 committed:
Replace Duke with Janet

wjunjie01 committed:
Add echo function

wjunjie01 committed:
Add record task list function

wjunjie01 committed:
Add mark task function in task list

wjunjie01 committed:
Add unmark task function

wjunjie01 committed:
Change the code to comply with coding standard

wjunjie01 committed:
Change the method name to comply with coding standard

Variables name should be Nouns.
ClassName should be PascalCase.
variableName should be camelCase.
CONSTANT_NAME should be in CAPS.

Method names should be a Verb.

wjunjie01 committed:
Add ToDo, Deadline, Event class

This is done using the inheritance function from the superclass Task.
Override in the function toString has been implemented to differentiate
the outputs between the classes.

wjunjie01 committed:
Add record function for the different task type: Todo, Deadline, Event

Able to display the task type along with its status.

wjunjie01 committed:
Add Ui class

Major refactoring of code part 1, improving code quality by cleaning
up code statements and adhering to code quality

wjunjie01 committed:
Add Parser class

Major refactoring of code part 2

wjunjie01 committed:
Clean up code quality

Minor indentation and spacing issues

wjunjie01 committed:
Add Command classes for easier abstraction on the Parser class

wjunjie01 committed:
Add TaskList class

This is to access the tasks from the task list easily and dynamically.

wjunjie01 committed:
Refactor the code for Parser and Duke class

This is done to make the flow in the main function more easy to follow.
This incorporates the error handling effectively.

wjunjie01 committed:
Merge branch 'branch-Level-5'

wjunjie01 committed:
Add automated text UI testing

awesomesjh commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+public class Task{
```
Perhaps a space after the class name? I noticed the same issue in several other places too.
```suggestion
public class Task {
```

awesomesjh commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,132 @@
+
+    public static Command parseEventTask(String input) throws DukeException {
+        String taskInformation = input.substring(BEGIN_EVENT_INDEX);
+        String[]fromSplitArray = taskInformation.split(" /from ");
```
Maybe a space after []?
```suggestion
        String[] fromSplitArray = taskInformation.split(" /from ");
```

awesomesjh commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,132 @@
+        String firstWord = input.split(" ")[0];
+
+        switch (firstWord) {
+        case "bye": {
+            return new ByeCommand();
+        }
```
Should the case statement have braces? I noticed the same issue in several other places too.
```suggestion
        case "bye":
            return new ByeCommand();
```

awesomesjh commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+    }
+    @Override
+    public String toString() {
+        return "[D]"+ super.toString() + " (by: " + by + ")";
```
Perhaps a space before +? I noticed the same issue in several other places too.
```suggestion
        return "[D]" + super.toString() + " (by: " + by + ")";
```

awesomesjh reviewed (commented):
Overall, I found that most of the code followed the Java coding standard. I noted a few coding violations related to spacing. Consider using consistent spacing throughout the codebase.

isaaceng7 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,132 @@
+        String[] taskInfoArray = taskInformation.split(" /by ");
+
+        if (taskInfoArray[0].trim().isEmpty()){
+            throw new DukeException("Your deadline task cannot be empty! "
```
Could create a method for this

isaaceng7 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+        this.description = description;
+        this.isDone = false;
+        taskCount += 1;
+    }
+    public String getStatus(){
```
be consistent with the spacing

isaaceng7 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+    }
+}
+
+//Inheritance and Polymorphism
```
Can put deadline, todo, event in different files

isaaceng7 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+    }
+    @Override
+    public String toString() {
+        return "[E]"+ super.toString() + " (from: " + from + " to: " + to + ")";
```
be consistent with spacing "[E]"_+ super.toString()...

wjunjie01 committed:
Improve code quality after reviewing comments on my first PR review

wjunjie01 committed:
Organize the classes into different packages

wjunjie01 committed:
Merge branch 'branch-A-Packages'

wjunjie01 committed:
Add delete task function

Add delete function and fix some bugs with regards to an invalid index
mark 0 or delete 0 that crashes the program.

json-bot-boop commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,8 @@
+package duke;
+
+public class DukeException extends Exception{
+    public DukeException(String errorMessage){
```
Leave a space before `{`

json-bot-boop commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,8 @@
+
+public class DukeException extends Exception{
+    public DukeException(String errorMessage){
+        //Constructor of super class, exception that prints the error message
```
Do not need to comment information that is explained in the code

json-bot-boop commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,130 @@
+        String[] inputArray = input.split(" ");
+
+        if (inputArray.length < 2) {
+            throw new DukeException("Please specify a task number to mark! Correct usage: mark [duke.task.Task Number]");
```
Try to avoid magic string

json-bot-boop commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,130 @@
+        int taskIndex;
+
+        try {
+            taskIndex = Integer.parseInt(inputArray[1]);
```
Try to avoid magic number

json-bot-boop commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,130 @@
+        String startDate = toSplitArray[0].trim();
+        String endDate = toSplitArray[1].trim();
+
+        if (startDate.isEmpty() || endDate.isEmpty() || taskDescription.isEmpty()) {
```
Try not to have complicated expression.

json-bot-boop commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,60 @@
+        System.out.println("Boo! This task is not done yet! ");
+        System.out.println(taskList.taskList.get(taskIndex).toString());
+        System.out.println(LINE);
+    }
```
Try to abstract the method so to not repeat yourself

wjunjie01 committed:
Create storage class to load data and save data from the file

wjunjie01 committed:
Add methods to the task classes to convert to file format

wjunjie01 committed:
Add save file and load from file function

File path is given by data/duke.txt

wjunjie01 committed:
Merge branch 'branch-Level-6'

wjunjie01 committed:
Merge branch 'branch-Level-7'

wjunjie01 committed:
Add JAR manifest

json-bot-boop reviewed (commented):
Overall, the code is okay except some code quality and coding standard issues.

wjunjie01 committed:
Add comments for Parser and Storage class

wjunjie01 committed:
Add comments to Command classes

wjunjie01 committed:
Modify code to align with code in A-MoreOOP

This is done by making the Command class abstract and
simplifying the code in Duke.java.

wjunjie01 committed:
Add comments to Task and its child classes

wjunjie01 committed:
Add find command

wjunjie01 committed:
Merge pull request #1 from wjunjie01/branch-Level-9

Add find command

wjunjie01 committed:
Merge branch 'master' into A-JavaDoc

wjunjie01 committed:
Merge pull request #2 from wjunjie01/A-JavaDoc

Add JavaDoc comments

wjunjie01 committed:
Clean up code after merging

- Add better print statements
- Remove unnecessary methods and variable
- Add missing comments to merged methods

wjunjie01 committed:
Clean up code to adhere to coding standards

- Mainly the bracket ({) that is supposed to have a spacing after the
function

wjunjie01 committed:
Clean up UI class

- Abstract out methods
- Fix print statements and error message

wjunjie01 committed:
Fix comments for all classes

- Add comments for find command
- Fix comments for all the parameters of the classes
- Remove unnecessary variable and method for Command class
- Change Todo class's behavior

wjunjie01 committed:
Add basic User Guide(UG)

- Includes the basic features and structure
- Change find command's search to be case insensitive
- Some refinement on Parser class

wjunjie01 committed:
Update FAQ, Command summary section of UG

wjunjie01 committed:
Update README.md to include warnings

wjunjie01 committed:
Update README.md change from IP to Janet

wjunjie01 committed:
Add new picture for Janet.jar for README

wjunjie01 committed:
Update README to be compatible with the webpage
